### PR TITLE
Zalgo updates

### PR DESCRIPTION
**Pull request by emmettu**
Simplify Zalgo algorithm:
For each character in input message, insert a diacritic if the random
float generated is less than the Z_RATE.

### CODE REVIEW

**Comment by emmettu** [comment]
@yuqi-zhang please review.

File: vilebot/src/com/oldterns/vilebot/util/Zalgo.java [hunk 1]
```diff
@@ -77,44 +72,26 @@
...
-            else if (nextCharDirection == 3) {
-                result.append(UP_CHARS[rand.nextInt(UP_CHARS.length)]);
-                result.append(MID_CHARS[rand.nextInt(MID_CHARS.length)]);
-                result.append(DOWN_CHARS[rand.nextInt(DOWN_CHARS.length)]);
-            }
+    private static String getZalgo(char c) {
+        if (rand.nextFloat() < Z_RATE) {
+            return c + randomDia();
         }
+        return "" + c;
```
Comment (yuqi-zhang) [inline vilebot/src/com/oldterns/vilebot/util/Zalgo.java]: Does this serve any other purpose?

Comment (emmettu) [inline vilebot/src/com/oldterns/vilebot/util/Zalgo.java]: Converts to from char to string.

**Comment by yuqi-zhang** [comment]
LGTM, but does this fix the broken chars issue?

**Comment by emmettu** [comment]
I don't actually know if it fixes broken chars I tested it with some random phrases and it looked OK.

**Comment by yuqi-zhang** [comment]
I think some of the phrases !speak spews out have weird characters in them. In any case I'm good with merging this and seeing how it works in #thefoobar

**Comment by lberk** [comment]
Emmett Underhill <notifications@github.com> writes:
> I don't actually know if it fixes broken chars I tested it with some
> random phrases and it looked OK.

Perhaps it's time for a proper testsuite

**Comment by emmettu** [comment]
IMO A testsuite would not be _idiomatic_ VileBot

**Pull request merged**